src/mbgl/text/glyph_pbf.cpp

```diff
@@ -23,68 +21,96 @@ std::vector<Glyph> parseGlyphPBF(const GlyphRange& glyphRange, const std::string
             // needs to satisfy a few metrics conditions that ensure that the glyph isn't bogus.
             // All other glyphs are malformed.  We're also discarding all glyphs that are outside
             // the expected glyph range.
-            if (!hasID || !hasWidth || !hasHeight || !hasLeft || !hasTop || !hasAdvance ||
-                glyph.metrics.width >= 256 || glyph.metrics.height >= 256 ||
-                glyph.metrics.left < -128 || glyph.metrics.left >= 128 ||
-                glyph.metrics.top < -128 || glyph.metrics.top >= 128 ||
-                glyph.metrics.advance >= 256 ||
+            if (!hasID || !hasWidth || !hasHeight || !hasLeft || !hasTop || !hasAdvance || glyph.metrics.width >= 256 ||
+                glyph.metrics.height >= 256 || glyph.metrics.left < -128 || glyph.metrics.left >= 128 ||
+                glyph.metrics.top < -128 || glyph.metrics.top >= 128 || glyph.metrics.advance >= 256 ||
                 glyph.id < glyphRange.first || glyph.id > glyphRange.second) {
-                continue;
+                return;
             }
 
             // If the area of width/height is non-zero, we need to adjust the expected size
             // with the implicit border size, otherwise we expect there to be no bitmap at all.
             if (glyph.metrics.width && glyph.metrics.height) {
-                const Size size {
-                    glyph.metrics.width + 2 * Glyph::borderSize,
-                    glyph.metrics.height + 2 * Glyph::borderSize
-                };
+                const Size size{glyph.metrics.width + 2 * Glyph::borderSize,
+                                glyph.metrics.height + 2 * Glyph::borderSize};
 
                 if (size.area() != glyphData.size()) {
-                    continue;
+                    return;
                 }
 
                 glyph.bitmap = AlphaImage(size, reinterpret_cast<const uint8_t*>(glyphData.data()), glyphData.size());
             }
 
             result.push_back(std::move(glyph));
+        };
+
+        double ascender{0.0}, descender{0.0};
+        uint16_t count{0};
+        auto fontstack_pbf = glyphs_pbf.get_message();
+        while (fontstack_pbf.next()) {
+            switch (fontstack_pbf.tag()) {
+                case 3: {
+                    readGlyphMetrics(fontstack_pbf);
+                    ++count;
+                    break;
+                }
+                case 4: {
+                    ascender = fontstack_pbf.get_double();
+                    break;
+                }
+                case 5: {
+                    descender = fontstack_pbf.get_double();
+                    break;
+                }
+                default: {
+                    fontstack_pbf.skip();
+                    break;
+                }
+            }
+        }
+        if (ascender != 0.0 || descender != 0.0) {
+            assert(count <= result.size());
+            for (uint16_t i = result.size() - count; i <= result.size() - 1; ++i) {
```

Comment:
Better to cache `result.size() - 1`, otherwise it's re-calculated at each iteration.